### ISSUE TITLE
Support Reboot Scenarios

### ISSUE THREAD

**Issue by nibanks** [microsoft/azure-pipelines-agent#3540]
This is related to https://github.com/microsoft/azure-pipelines-agent/issues/1505, but since that's been closed a while and it seemed specific to Release scenarios (which this Issue is not) I figured I'd open a new issue. Also related to https://github.com/actions/virtual-environments/issues/3858.

## Background

Significant work in the Microsoft Windows team has started undocking/open-sourcing internal Windows components, particularly drivers. We recently got test signing enabled on the machines (https://github.com/actions/virtual-environments/issues/1637) and we can finally start installing and running tests for these drivers. The next major requirement that's coming out of this is the ability debug/diagnose issues when testing Windows drivers; particularly when the machine bug checks. This is an unfortunate result of normal driver testing; things crash and the machine reboots. Currently, when this happens though, all state is lost and even when the agent restarts, there's no way for us to collect the crash dump that was created.

## Ask

We require the ability for our jobs to survive a reboot of the agent so that we can run additional tasks to collect state (mostly the memory.dmp file) to upload to the artifacts for later diagnosis by the developer.

**Comment by nibanks** [comment]
I see this was added to the backlog. Is there any ETA that is associated with that? Any way to possibly bump the priority on this, as we're dependent on it for internal Windows development?

**Comment by nibanks** [comment]
Ping. Any updates on this? Is there someone I should be talking to get some priority on this? @anatolybolshakov?

**Comment by arsdragonfly** [comment]
Our team is working on Windows updates validation scenarios and would also need this feature.

**Comment by adstep** [comment]
My team also does work around interactive testing for applications on Windows that requires reboots to be applied to the device. Am also interested in getting support for this feature.

**Comment by sumit84mca** [comment]
This is very required request. I also need to install application and then reboot on deployment group agents.

**Comment by Darth-Shader** [comment]
Ping. Is this issue addressed on the new prerelease? Who is the current owner of this feature?
Thanks.

**Comment by nibanks** [comment]
@max-zaytsev what's the next step here? How do we make progress? We've significantly increased our kernel mode usage/testing with GitHub actions and Azure Pipelines since this issue was opened, and our need has only grown. Additionally, we plan to further expand soon.

**Comment by ligei2** [comment]
This would be a useful feature for being able to retry jobs, which are canceled when the agent reboots.

**Comment by arsdragonfly** [comment]
@anatolybolshakov can we re-open this issue and move it back to the backlog?

**Comment by CN120** [comment]
Would really love this feature to be added. My company deals with on prem software that needs to be installed on agents and rebooted before we can run tests. Trying to deal with remote orchestration of all of this is a total headache and takes up way more resources.
